spec/lib/tmuxinator/window_spec.rb

```diff
@@ -158,27 +159,72 @@
     end
   end
 
+  describe "#name_options" do
+    context "with a name" do
+      let(:window_name) { "editor" }
+
+      it "specifies name with tmux name option" do
+        expect(window.tmux_window_name_option).to eq "-n #{window_name}"
+      end
+    end
+
+    context "without a name" do
+      let(:window_name) { nil }
+
+      it "specifies no tmux name option" do
+        expect(window.tmux_window_name_option).to be_empty
+      end
+    end
+  end
+
   describe "#tmux_new_window_command" do
     let(:project) { double(:project) }
     let(:window) { Tmuxinator::Window.new(yaml, 0, project) }
+    let(:root?) { true }
+    let(:root) { "/project/tmuxinator" }
 
     before do
       allow(project).to receive_messages(
-        :name => "",
+        :name => "test",
         :tmux => "tmux",
-        :root => "/project/tmuxinator",
-        :root? => true,
+        :root => root,
+        :root? => root?,
         :base_index => 1
       )
     end
 
-    context "tmux 1.6 and below" do
-      before do
-        allow(Tmuxinator::Config).to receive_messages(:version => 1.6)
+    let(:tmux_part) { project.tmux }
+    let(:window_command_part) { "new-window" }
+    let(:name_part) { window.tmux_window_name_option }
+    let(:target_part) { "-t #{window.tmux_window_target}" }
+    let(:path_part) { "#{default_path_option} #{project.root}" }
+
+    let(:default_path_option) { '-c' }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.